.semaphore/semaphore.yml

```diff
@@ -0,0 +1,133 @@
+          commands:
+            - ${MVN} test-compile ${MAVEN_SKIP} ${MAVEN_SKIP_TESTS}
+            - ${MVN} forbiddenapis:check forbiddenapis:testCheck --fail-at-end
+
+        - name: "pmd checks"
+          commands:
+            - ${MVN} pmd:check --fail-at-end  # TODO: consider adding pmd:cpd-check
+
+        - name: "spotbugs checks"
+          commands:
+            - ${MVN} spotbugs:check --fail-at-end -pl '!benchmarks'
+
+        - name: "analyze dependencies"
+          commands:
+            - >
+              ${MVN} ${MAVEN_SKIP} dependency:analyze -DoutputXML=true -DignoreNonCompile=true -DfailOnWarning=true --fail-at-end || { echo "
+
+              The dependency analysis has found a dependency that is either:
+              1) Used and undeclared: These are available as a transitive dependency but should be explicitly
+              added to the POM to ensure the dependency version. The XML to add the dependencies to the POM is
+              shown above.
+              2) Unused and declared: These are not needed and removing them from the POM will speed up the build
+              and reduce the artifact size. The dependencies to remove are shown above.
+              If there are false positive dependency analysis warnings, they can be suppressed:
+              https://maven.apache.org/plugins/maven-dependency-plugin/analyze-mojo.html#usedDependencies
+              https://maven.apache.org/plugins/maven-dependency-plugin/examples/exclude-dependencies-from-dependency-analysis.html
+              For more information, refer to:
+              https://maven.apache.org/plugins/maven-dependency-plugin/analyze-mojo.html
+              " && false; }
+
+        - name: "Confluent Extensions"
+          env_vars:
+            - name: MAVEN_PROJECTS
+              value: extensions-contrib/confluent-extensions
+          commands: &run_tests
+            - >
+              MAVEN_OPTS="${MAVEN_OPTS} -Xmx1g" ${MVN} test -pl ${MAVEN_PROJECTS}
+              ${MAVEN_SKIP} -Dremoteresources.skip=true
+
+        - name: "Server"
+          env_vars:
+            - name: MAVEN_PROJECTS
+              value: server
+          commands: *run_tests
+
+        - name: "Processing"
+          env_vars:
+            - name: MAVEN_PROJECTS
+              value: processing
+          commands: *run_tests
+
+        - name: "Indexing Service"
+          env_vars:
+            - name: MAVEN_PROJECTS
+              value: indexing-service
+          commands: *run_tests
+
+        - name: "Kafka Indexing Service"
+          env_vars:
+            - name: MAVEN_PROJECTS
+              value: extensions-core/kafka-indexing-service
+          commands: *run_tests
+
+        - name: "Other Tests"
+          env_vars:
+            - name: MAVEN_PROJECTS
+              value: '!server,!processing,!indexing-service,!extensions-core/kafka-indexing-service,!extensions-contrib/confluent-extensions,!integration-tests-ex/cases'
```

Comment:
There's a new extension added for running kubernetes tasks as jobs, should we add that here or is it taken care somewhere else?

Reply:
@saithal-confluent won't that automatically be included here? To me it looks like we are excluding projects from here which already have their own dedicated job. Right?